artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ArtemisExecutor.java

```diff
@@ -50,6 +50,16 @@ default int shutdownNow(Consumer<? super Runnable> onPendingTask) {
       return 0;
    }
 
+   default boolean flush(long timeout, TimeUnit unit) {
+      CountDownLatch latch = new CountDownLatch(1);
+      execute(latch::countDown);
```

Comment:
if the latch is submitted from within the same thread that is executing the tasks, waiting won't be necessary, because it cannot be triggered

Reply:
This won’t be used at all.  It’s just the default implementation for tests.  OrderedExecutor has an implementation.